internal/storage/v2/elasticsearch/tracestore/from_dbmodel.go

```diff
@@ -52,14 +100,18 @@ func dbProcessToResource(process dbmodel.Process, resource pcommon.Resource) {
 	} else {
 		attrs.EnsureCapacity(len(tags))
 	}
+	tags = f.mergeTagMapAndTagSlice(tags, tag)
 	dbTagsToAttributes(tags, attrs)
 }
 
-func dbSpansToSpans(dbSpans []dbmodel.Span, resourceSpans ptrace.ResourceSpansSlice) error {
+func (f *fromDBModel) dbSpansToSpans(dbSpans []dbmodel.Span, resourceSpans ptrace.ResourceSpansSlice) error {
 	for i := range dbSpans {
 		span := &dbSpans[i]
+		tags := f.mergeTagMapAndTagSlice(span.Tags, span.Tag)
+		span.Tag = nil
+		span.Tags = tags
 		resourceSpan := resourceSpans.AppendEmpty()
-		dbProcessToResource(span.Process, resourceSpan.Resource())
+		f.dbProcessToResource(span.Process, resourceSpan.Resource())
 
 		scopeSpans := resourceSpan.ScopeSpans()
 		scopeSpan := scopeSpans.AppendEmpty()
```

Comment:
> dbSpanToScope(span, scopeSpan)

this is misleading - you don't need the `span` object to populate a `scopeSpan`, you only need to have a combine map of tags.